modules/cvv/src/stfl/stfl_engine.hpp

```diff
@@ -22,6 +22,17 @@
 #include "element_group.hpp"
 #include "../qtutil/util.hpp"
 
+
+// WORKAROUND:
+//   - Qt::SplitBehavior introduced in 5.14, QString::SplitBehavior was removed in Qt6
+//   - Support required part of Qt::SplitBehavior from 5.15 for older Qt versions
+#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
+namespace Qt {
+	static constexpr QString::SplitBehavior SkipEmptyParts = QString::SkipEmptyParts;
```

Comment:
There is compilation failure on Ubuntu 16.04 with Qt 5:

```
--   GUI:                           QT5
--     QT:                          YES (ver 5.5.1 )
--       QT OpenGL support:         YES (Qt5::OpenGL 5.5.1)
```

```
In file included from /build/precommit-contrib_custom_linux64/4.x/opencv_contrib/modules/cvv/src/impl/../view/rawview.hpp:15:0,
                 from /build/precommit-contrib_custom_linux64/4.x/opencv_contrib/modules/cvv/src/impl/init.cpp:22:
/build/precommit-contrib_custom_linux64/4.x/opencv_contrib/modules/cvv/src/impl/../view/../stfl/stfl_engine.hpp: In member function 'QStringList cvv::stfl::STFLEngine<Element>::getSuggestionsForCmdQuery(const QString&, size_t)':
/build/precommit-contrib_custom_linux64/4.x/opencv_contrib/modules/cvv/src/impl/../view/../stfl/stfl_engine.hpp:776:46: error: there are no arguments to 'qsizetype' that depend on a template parameter, so a declaration of 'qsizetype' must be available [-fpermissive]
        std::min(qsizetype(hasByString ? 2 : 1), qsizetype(tokens.size()));
                                              ^
/build/precommit-contrib_custom_linux64/4.x/opencv_contrib/modules/cvv/src/impl/../view/../stfl/stfl_engine.hpp:776:46: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
/build/precommit-contrib_custom_linux64/4.x/opencv_contrib/modules/cvv/src/impl/../view/../stfl/stfl_engine.hpp:776:72: error: there are no arguments to 'qsizetype' that depend on a template parameter, so a declaration of 'qsizetype' must be available [-fpermissive]
        std::min(qsizetype(hasByString ? 2 : 1), qsizetype(tokens.size()));
                                                                        ^
```

http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_custom_linux64/builds/100000

Reply:
Support for Ubuntu 16.04 LTS ended more than a year ago.

Ubuntu 16.04 ESM has support until 2026 and comes with QT 5.5 (EOL 2018). Ubuntu 14.04 ESM has support until 2024 and comes with QT 5.2 (EOL ~2015?). RHEL 6 has support until 2024 and comes with QT 4.6 (EOL ~2012?).

My personal opinion is that we should not support ESM releases at all. Open source LTS distros have a relatively large user base, but commercial ESM distros have few users. I think among them there are very few users who want to combine the latest version of OpenCV with the ancient Qt version from package management.

If support for commercial extended support releases is nevertheless desired, I need a list of which ones.

I'm getting more and more calls for help from colleagues because Qt6 doesn't work, even though I understand that backwards compatibility is also an important point.

Reply:
In general, patch should not introduce regressions. Even for "unsupported" build configurations.

Please apply this patch to avoid build error:

```.diff
--- a/modules/cvv/src/stfl/stfl_engine.hpp
+++ b/modules/cvv/src/stfl/stfl_engine.hpp
@@ -773,7 +773,7 @@ private:
                if (cmd == "group" || cmd == "sort")
                {
                        int frontCut =
-                           std::min(qsizetype(hasByString ? 2 : 1), qsizetype(tokens.size()));
+                           std::min(size_t(hasByString ? 2 : 1), size_t(tokens.size()));
                        tokens = cmdQuery.split(" ", Qt::SkipEmptyParts)
                                     .mid(frontCut, tokens.size());
                        QStringList args = tokens.join(" ").split(
```